tests/unit/browser/webengine/test_webengine_cookies.py

```diff
@@ -0,0 +1,73 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015-2019 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+from PyQt5.QtCore import QUrl
+from unittest.mock import MagicMock
+from qutebrowser.browser.webengine.cookies import _accept_cookie
+from qutebrowser.utils.urlmatch import UrlPattern
+
+
+def test_accept_cookie(config_stub):
+    """Test that _accept_cookie respects content.cookies.accept."""
+    request = MagicMock()
```

Comment:
Why use a mock at all? Like you've seen, that allows getting attributes which were never defined, which can be troublesome.

Instead I'd use the real thing, but move it to a [pytest fixture](https://docs.pytest.org/en/latest/fixture.html) so the tests are skipped on older Qt versions where cookie filters aren't available. Something like this (untested):

```python
import pytest
from PyQt5.QtWebEngineCore import QWebEngineCookieStore

@pytest.fixture
def filter_request():
    try:
        return QWebEngineCookieStore.FilterRequest()
    except AttributeError:
        pytest.skip("FilterRequest not available")
```

Then you can just add a `filter_request` argument to tests which need one.

(Note: Don't call it `request` because that's [a special pytest fixture](https://docs.pytest.org/en/latest/reference.html#request))